IPython/core/completer.py

```diff
@@ -2073,14 +2069,22 @@ def _complete(self, *, cursor_line, cursor_pos, line_buffer=None, text=None,
         return text, _matches, origins, completions
         
     def fwd_unicode_match(self, text:str) -> Tuple[str, list]:
+        if self._names is None:
```

Comment:
Do you really need to make this check?

Reply:
Yes, so that it is recomputed only once. 
Above it is set to `None`, and the first time this is called `self._names` get computed.